packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -95,6 +105,28 @@ export async function executePlugin(
   };
 }
 
+const wrapProgress = async (
```

Comment:
I know this extra wrapper is ugly. A 100% clean implementation I would prio when the new progress animations for the CLI land.